packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1523,10 +2010,130 @@ fn _run(
                     }
                 }
             }
+
             Ok(())
         }.in_current_span();
     };
 
     let local = tokio::task::LocalSet::new();
     local.block_on(&runtime, impl_future)
 }
+
+/// Helper function to create a [v8::String]
+fn new_js_string<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    s: impl AsRef<str>,
+) -> JSRunnerResult<v8::Local<'s, v8::String>> {
+    let s = s.as_ref();
+    v8::String::new(scope, s)
+        .ok_or_else(|| JSRunnerError::V8(format!("Could not create String: {s}")))
+}
+
+/// C representation of Arrow array data nodes
+#[repr(C)]
+#[derive(Debug)]
+pub struct DataFfi<'s> {
```

Comment:
From this struct onwards it may make sense to move the content to another file.

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9